### PR TITLE
CMake cleanup

### PR DESCRIPTION
**Pull request by juan-lunarg**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -5,22 +5,16 @@ enable_testing()
 
 set(CMAKE_OSX_DEPLOYMENT_TARGET "10.12" CACHE STRING "Minimum OS X deployment version")
 
-project(VULKAN_PROFILES LANGUAGES CXX C)
+project(VULKAN_PROFILES LANGUAGES CXX)
```
Comment (juan-lunarg) [inline CMakeLists.txt:8]: It's best practice to minimize the compilers used for your project. It complicates the CMake code.

File: CMakeLists.txt [hunk 2]
```diff
@@ -5,22 +5,16 @@ enable_testing()
...
 # Regenerating profiles requires Python 3.7.2 or better (for typing.OrderedDict).
 # Regeneration jsoncpp requires Python 3.
 # Otherwise Python is not required, as checked-in generated code and
 # headers will be used.
 option(PROFILES_BUILD_TESTS "Build profile tests" ON)
 
-set(CMAKE_ARCHIVE_OUTPUT_DIRECTORY "${CMAKE_BINARY_DIR}/lib" CACHE PATH "Archive output dir.")
-set(CMAKE_LIBRARY_OUTPUT_DIRECTORY "${CMAKE_BINARY_DIR}/lib" CACHE PATH "Library output dir.")
-set(CMAKE_PDB_OUTPUT_DIRECTORY     "${CMAKE_BINARY_DIR}/bin" CACHE PATH "PDB (MSVC debug symbol)output dir.")
-set(CMAKE_RUNTIME_OUTPUT_DIRECTORY "${CMAKE_BINARY_DIR}/bin" CACHE PATH "Executable/dll output dir.")
```
Comment (juan-lunarg) [inline CMakeLists.txt:19]: This was needed for the jsoncpp but isn't actually good practice. Now that jsoncpp is built standalone and consumed via find_package we don't need to do this anymore.

File: CMakeLists.txt [hunk 3]
```diff
@@ -5,22 +5,16 @@ enable_testing()
...
-
 set(CMAKE_POSITION_INDEPENDENT_CODE ON)
 set(CMAKE_CXX_VISIBILITY_PRESET "hidden")
-set(CMAKE_C_VISIBILITY_PRESET "hidden")
```
Comment (juan-lunarg) [inline CMakeLists.txt:23]: Now that we only specify CXX for the language we can remove CMake code like this.

File: CMakeLists.txt [hunk 4]
```diff
@@ -139,11 +133,6 @@ set(CMAKE_CXX_EXTENSIONS OFF)
 # Define GNU standard installation directories.
 include(GNUInstallDirs)
 
-if(WIN32 AND CMAKE_INSTALL_PREFIX_INITIALIZED_TO_DEFAULT)
-    # Windows: if install locations not set by user, set install prefix to "<build_dir>\install".
-    set(CMAKE_INSTALL_PREFIX "${CMAKE_BINARY_DIR}/install" CACHE PATH "default install path" FORCE)
-endif()
-
```
Comment (juan-lunarg) [inline CMakeLists.txt:146]: This was removed from vvl/loader a bit ago. It's not needed and can cause users issues.

File: CMakeLists.txt [hunk 5]
```diff
@@ -157,29 +146,24 @@ if (BUILD_WERROR)
 endif()
 
 # Platform-specific compiler switches
-if(${CMAKE_C_COMPILER_ID} STREQUAL "Clang")
-    add_compile_options(-Werror)
-endif()
-if(${CMAKE_C_COMPILER_ID} MATCHES "(GNU|Clang)")
-    add_compile_options(-Werror)
```
Comment (juan-lunarg) [inline CMakeLists.txt:164]: BUILD_WERROR handles this.

File: layer/CMakeLists.txt [hunk 6]
```diff
@@ -45,11 +45,10 @@ add_definitions(-DVK_ENABLE_BETA_EXTENSIONS)
 set(TARGET_NAME khronos_profiles)
 
 if (WIN32)
-    set (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -D_CRT_SECURE_NO_WARNINGS /bigobj")
-    set (CMAKE_C_FLAGS   "${CMAKE_C_FLAGS} -D_CRT_SECURE_NO_WARNINGS /bigobj")
+    add_compile_definitions(_CRT_SECURE_NO_WARNINGS)
+    add_compile_options(/bigobj)
 else()
-    set (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wpointer-arith")
-    set (CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wpointer-arith")
```
Comment (juan-lunarg) [inline layer/CMakeLists.txt:52]: 2 things.

1.) CMAKE_C_FLAGS isn't needed at all anymore.
2.) Unless you HAVE to don't touch these variables.

File: CMakeLists.txt (hunk 2 above)
Comment (juan-lunarg) [reply CMakeLists.txt:19]: This is one of the benefits of using find_package versus add_subdirectory.

add_subdirectory makes you have to care about the quality of the CMake code of the libraries you consume. Versus find_package just uses the already built artifacts.

File: CMakeLists.txt (hunk 5 above)
Comment (juan-lunarg) [reply CMakeLists.txt:164]: Also CMAKE_C_COMPILER_ID doesn't exist anymore. Now it's just CMAKE_CXX_COMPILER_ID

File: CMakeLists.txt (hunk 4 above)
Comment (ziga-lunarg) [reply CMakeLists.txt:146]: What is the default without this?

File: scripts/known_good.cmake [hunk 7]
```diff
@@ -1,12 +0,0 @@
...
-set(GIT_VULKAN_HEADER_REPOSITORY https://github.com/KhronosGroup/Vulkan-Headers)
-set(GIT_VULKAN_HEADER_COMMIT v1.3.239)
-
-# Vulkan Loader
-set(GIT_VULKAN_LOADER_REPOSITORY https://github.com/KhronosGroup/Vulkan-Loader)
-set(GIT_VULKAN_LOADER_COMMIT v1.3.239)
-
-# Vulkan Validation Layer
-set(GIT_VULKAN_VALIDATION_REPOSITORY https://github.com/KhronosGroup/Vulkan-ValidationLayers)
-set(GIT_VULKAN_VALIDATION_COMMIT v1.3.239)
```
Comment (ziga-lunarg) [inline scripts/known_good.cmake:11]: Where will this be set now?

File: CMakeLists.txt (hunk 4 above)
Comment (juan-lunarg) [reply CMakeLists.txt:146]: Not sure. The main issue is it prevents users/ci from setting [CMAKE_PREFIX_PATH](https://cmake.org/cmake/help/latest/envvar/CMAKE_PREFIX_PATH.html) as an environment variable.

File: scripts/known_good.cmake (hunk 7 above)
Comment (juan-lunarg) [reply scripts/known_good.cmake:11]: This is set in the known_good.json

**Review by ziga-lunarg — APPROVED** [review]
Good cleanup and helpful comments

File: CMakeLists.txt (hunk 4 above)
Comment (juan-lunarg) [reply CMakeLists.txt:146]: Also it breaks the new `--install` functionality which is also quite useful for package managers (and local users)

```
cmake --install build --prefix build/install
```

File: CMakeLists.txt (hunk 2 above)
Comment (juan-lunarg) [reply CMakeLists.txt]: Looks like profiles maybe depending on the layout of the build directory so I reverted this change for now.

**Comment by juan-lunarg** [comment]
Looks like the Android CI is having some issues unrelated to this PR

**Comment by juan-lunarg** [comment]
Android CI issues

**Comment by lunarpapillo** [comment]
?  Was this cherry-picked, or abandoned?

**Comment by juan-lunarg** [comment]
Abandoned for now

**Pull request closed without merging**